### DISCUSSION TITLE
Error of depth, clip_vision and normal_map in SD_Preprocessor

### DISCUSSION THREAD

**Discussion by supwang** [stassius/StableHoudini#25] [Q&A]
Hi,
I got error of depth and clip_vision in SD_Preprocessor. Could you give a help? Thank you.

Error message of depth:
<img width="559" alt="1" src="https://user-images.githubusercontent.com/4122964/236682223-d611e965-9b10-4e8a-8e0c-43538b51ed06.png">
<img width="865" alt="2" src="https://user-images.githubusercontent.com/4122964/236682256-8722282e-dd18-4d95-93f6-4fe026b31835.png">

Error image of clip_vision:
<img width="1237" alt="3" src="https://user-images.githubusercontent.com/4122964/236682280-98bb3a85-d1a5-45f6-88c3-bce123148901.png">

Error image of normal_map:
<img width="1479" alt="4" src="https://user-images.githubusercontent.com/4122964/236682533-9afd64f1-95e1-4e89-a97d-e51c7dfa4819.png">

**Comment by stassius** [comment]
Clip vision doesn't output an image, so it won't work. Check if Depth and Normal are working in webui - you can turn on preview and test it.
![image](https://user-images.githubusercontent.com/35260274/236684661-6e1d96c3-fd50-49ba-937c-e5e1d8860cc4.png)

**Reply by supwang** [reply]
1. depth_leres, depth_leres++, depth_midas, normal_bae and normal_midas can be used in stable_diffusion_webui.
2. depth_leres, depth_leres++ are working fine in Houdini, depth got error
3. normal_map got error in Houdini
4. how to show the result of clip_vision in Houdini?

**Reply by supwang** [reply]
> Clip vision doesn't output an image, so it won't work. Check if Depth and Normal are working in webui - you can turn on preview and test it.

I used the last hda files of May.9. Now the depth can be shown normally, but the result of normal_map seems got error.
Original image:
![00005](https://github.com/stassius/StableHoudini/assets/4122964/e6dd9087-fbf4-4c49-9201-ed50cdebf386)
normal_map image:
![detect_0](https://github.com/stassius/StableHoudini/assets/4122964/41c00c73-acfc-4582-b418-bbe97b1d17bd)

**Comment by stassius** [comment]
Clip vision doesn't give you any image at all. It gives an embedding for use with t2i style adapter. normal_map is an outdated preprocessor, use normal_bar and normal_midas instead.

**Reply by supwang** [reply]
Thank you.
There have no normal_bar and normal_midas in Preprocessor, but normal_bae works normally.
<img width="176" alt="2023-05-10 182846" src="https://github.com/stassius/StableHoudini/assets/4122964/1d5983ea-6362-42c3-900f-99d986ae4c43">
